docs/developers/contributing.md

```diff
@@ -0,0 +1,171 @@
+comment along with the specific error code (e.g. `import sys  # noqa: E402`) but
+please do not ignore errors lightly.
+
+## Translations
+
+Starting with version 0.4.7, napari codebase include internationalization
+(i18n) and now offers the possibility of installing language packs, which
+provide localization (l10n) enabling the user interface to be displayed in
+different languages.
+
+To learn more about the current languages that are in the process of
+translation, visit the [language packs repository](https://github.com/napari/napari-language-packs)
+
+To make your code translatable (localizable), please use the `trans` helper
+provided by the napari utilities.
+
+```python
+from napari.utils.translations import trans
+
+some_string = trans._("Localizable string")
+```
+
+To learn more, please see the [translations guide](https://napari.org/guides/stable/translations.html).
+
+## Making changes
+
+Create a new feature branch:
+```sh
+git checkout master -b your-branch-name
+```
+
+`git` will automatically detect changes to a repository.
+You can view them with:
+```sh
+git status
+```
+
+Add and commit your changed files:
+```sh
+git add my-file-or-directory
+git commit -m "my message"
+```
+
+## Tests
+
+We use unit tests and integration tests to ensure that
+napari-animation works as intended. Writing tests for new code is a critical part of
+keeping napari-animation maintainable as it grows.
+
+Check out the dedicated documentation on testing over at [napari.org](https://napari.org/developers/testing.html) that we recommend you
+read as you're working on your first contribution.
+
+### Help us make sure it's you
+
+Each commit you make must have a [GitHub-registered email](https://github.com/settings/emails)
+as the `author`. You can read more [here](https://help.github.com/en/github/setting-up-and-managing-your-github-user-account/setting-your-commit-email-address).
+
+To set it, use `git config --global user.email your-address@example.com`.
+
+## Keeping your branches up-to-date
+
+Switch to the `master` branch:
+```sh
+git checkout master
+```
+
+Fetch changes and update `master`:
+```sh
+git pull upstream master --tags
+```
+
+This is shorthand for:
+```sh
+git fetch upstream master --tags
+git merge upstream/master
+```
+
+Update your other branches:
+```sh
+git checkout your-branch-name
+git merge master
+```
+
+## Sharing your changes
+
+Update your remote branch:
+```sh
+git push -u origin your-branch-name
+```
+
+You can then make a [pull-request](https://guides.github.com/activities/forking/#making-a-pull-request) to `napari`'s `master` branch.
+
+## Building the docs
+
+From the project root:
+```sh
+make docs
+```
+
+The docs will be built at `docs/_build/html`.
+
+Most web browsers will allow you to preview HTML pages.
+Try entering `file:///absolute/path/to/napari/docs/_build/html/index.html` in your address bar.
+
+To read more about the docs, how they're organized, and built, read {ref}`docs-dev`.
```

Comment:
When I build the documentation it says this reference {ref}`docs-dev` is undefined. What do we need to do to fix that?

Reply:
I got this file from Alister's previous PR and didn't review properly. Will fix.